### PR TITLE
Add Chronicle Sessions

### PR DESCRIPTION
**Pull request by ilikesymmetry**
Implement Chronicle Sessions that take in logs, convert them to parquet files, and upload them to our s3 bucket.

### CODE REVIEW

File: appdev/chronicle.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+from fastparquet import write
+from pandas import DataFrame
+from boto3 import client
+from datetime import datetime
+
+BUCKET = 'appdev-register'
+PATH = '/tmp'
+
+class ChronicleSession:
+
```
Comment (kq-li) [inline appdev/chronicle.py:15]: prefer no newline after class

File: appdev/chronicle.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    def __init__(self, accessKey, secretKey, app, cacheSize=10):
+        self.s3 = client('s3',aws_access_key_id=accessKey,aws_secret_access_key=secretKey)
```
Comment (kq-li) [inline appdev/chronicle.py]: spaces after commas, use snake (underscore) case instead of camel case

File: appdev/chronicle.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    def log(self, eventName, event):
+        if eventName not in self.logMap:
+            self.logMap[eventName] = [event]
+            return
+
+        logs = self.logMap[eventName]
+        logs.append(event)
+        self.logMap[eventName] = logs
+        if len(logs)>=self.cacheSize:
```
Comment (kq-li) [inline appdev/chronicle.py]: spaces around `>=`

File: appdev/chronicle.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            self.writeLogs(eventName)
+
+    def writeLogs(self, eventName):
+        organized = dict()
+        for event in self.logMap[eventName]:
+            for key in event.keys():
+                if key not in organized.keys():
+                    organized[key]=[event[key]]
```
Comment (kq-li) [inline appdev/chronicle.py]: space around `=`

File: appdev/chronicle.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                else:
+                    organized[key].append(event[key])
+        df = DataFrame(organized)
+        fileName = str(datetime.now())+'.parquet'
+        write(PATH+'/'+fileName, df)
+        self.s3.upload_file(PATH+'/'+fileName, BUCKET, self.app+'/'+eventName+'/'+fileName)
```
Comment (kq-li) [inline appdev/chronicle.py]: spaces around all these +

**Review by kq-li — COMMENTED** [review]
looks good, some style things

**Comment by young-k** [comment]
are you using a linter? make sure to set that up (look into `pylint`)

**Comment by ilikesymmetry** [comment]
@kq-li made styling changes

**Comment by kq-li** [comment]
appdev convention is 2 space tabs, i believe -- once that's changed, rebase commits and i'll approve

**Comment by young-k** [comment]
when you rebase, can you also uppercase that first commit

**Review by kq-li — APPROVED** [review]
lgtm, make sure to rebase

**Comment by young-k** [comment]
@kq-li doesn't something look sus about this rebase, why are older commits here?

**Comment by kq-li** [comment]
that is weird, maybe rebased too far?

**Pull request merged**